pkg/monitor/datapath_debug.go

```diff
@@ -238,12 +239,12 @@ func (n *DebugMsg) DumpInfo(data []byte) {
 }
 
 // Dump prints the debug message in a human readable format.
-func (n *DebugMsg) Dump(prefix string) {
-	fmt.Printf("%s MARK %#x FROM %d DEBUG: %s\n", prefix, n.Hash, n.Source, n.Message())
+func (n *DebugMsg) Dump(prefix string, linkMonitor getters.LinkGetter) {
```

Comment:
^^ I briefly considered other ways to try to avoid passing this parameter down through these methods, like embedding it in the `DebugMsg`. My main concern was that I believe that `DebugMsg` is binary-compatible with the datapath BPF C structures that generate these notifications, so we can't just add another pointer somewhere here as this could potentially cause us to either inadvertently write to a field that we shouldn't (thereby messing up some message in the buffer), or plausibly interpet some random bytes as a pointer. This seemed like the simpler / safer option.

Reply:
The problem seems to be that `DebugMsg` is used as receiver for this method. We could use a structure that had 2 fields, a `DebugMsg` and a `linkMonitor`, then this new structure could have the `Dump` method instead.

Reply:
yeah as I understand it, this is the role that the `MonitorFormatter` is trying to fill. Arguably we should shuffle the code around to attach this to the MonitorFormatter and hide these parameters from going down through a couple of layers of functions. At this point though I don't see any real benefit to going through all that.